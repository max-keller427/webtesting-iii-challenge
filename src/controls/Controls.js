import React from "react";

const Controls = props => {
  const { locked, closed, toggleLocked, toggleClosed } = props;
  console.log(props);

  return (
    <div className="controls panel">
      <button
        disabled={!closed}
        onClick={toggleLocked}
        className="toggle-btn"
        data-testid="key"
      >
        {locked ? "Unlock Gate" : "Lock Gate"}
      </button>
      <button
        data-testid="handle"
        disabled={locked}
        onClick={toggleClosed}
        className="toggle-btn"
      >
        {closed ? "Open Gate" : "Close Gate"}
      </button>
    </div>
  );
};

export default Controls;
