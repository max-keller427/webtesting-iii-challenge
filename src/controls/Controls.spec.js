// Test away!
import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react";
import * as rtl from "@testing-library/react";
import "jest-dom/extend-expect";

afterEach(rtl.cleanup);

import Controls from "./Controls";

describe("<Controls/>", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<Controls />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should provide buttons to toggle the closed and locked states.", () => {
    const { getByText } = render(<Controls />);

    const lockGateButton = getByText(/lock gate/i);
    const closeGateButton = getByText(/close gate/i);

    expect(lockGateButton).toBeTruthy();
    expect(closeGateButton).toBeTruthy();
  });
  // no clue why this syntax doesnt work
  xit("buttons' text changes to reflect the state the door will be in if clicked", () => {
    const { getByTestId } = render(<Controls locked={false} closed={false} />);

    const closeGateButton = getByTestId("handle");
    const lockGateButton = getByTestId("key");
    console.log(closeGateButton);
    fireEvent.click(closeGateButton);
    fireEvent.click(lockGateButton);

    expect(closeGateButton).toHaveTextContent(/open gate/i);
    expect(lockGateButton).toHaveTextContent(/unlock gate/i);
  });

  it("Locked buttons text changes to reflect the state the door will be in if clicked", async () => {
    const buttons = render(<Controls locked={true} closed={true} />);
    const closebtn = buttons.getByTestId("handle");
    const lockbtn = buttons.getByTestId("key");
    expect(closebtn.textContent).toBe("Open Gate");
    expect(lockbtn.textContent).toBe("Unlock Gate");
    await fireEvent.click(closebtn);
    await fireEvent.click(lockbtn);
  });

  it("the lock toggle button is disabled if the gate is open", () => {
    const { getByTestId } = render(<Controls locked={false} closed={false} />);

    const lockButton = getByTestId("key");

    fireEvent.click(lockButton);

    expect(lockButton.disabled).toBe(true);
  });

  it("the open toggle button is disabled if the gate is locked", () => {
    const { getByTestId } = render(<Controls locked={true} closed={true} />);

    const openButton = getByTestId("handle");

    fireEvent.click(openButton);

    expect(openButton.disabled).toBe(true);
  });
});
