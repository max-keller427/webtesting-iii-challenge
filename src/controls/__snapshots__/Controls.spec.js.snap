// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls/> matches the snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    data-testid="key"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    data-testid="handle"
  >
    Close Gate
  </button>
</div>
`;
